python/cudaq/_cudaq.cpp

```diff
@@ -56,6 +59,21 @@ PYBIND11_MODULE(_pycudaq, mod) {
   mpiSubmodule.def(
       "initialize", []() { cudaq::mpi::initialize(); },
       "Initialize MPI if available.");
+  mpiSubmodule.def(
```

Comment:
If you search for `.. doxygenfunction:: cudaq::mpi::initialize()`, could you just add the new functions in the same place so that they are documented?

Reply:
Oh, and I realized that in addition to the new gather_all function needing to be added to the C++ api code, the cudaq mpi python module is not documented. 
Search for `.. automethod:: cudaq::initialize_cudaq`; I suggest to just add them there with `.. automethod:: cudaq::mpi::method_name`. That should work without issue, and otherwise I can look into it.